src/Managers/BaseManager.php

```diff
@@ -64,179 +73,165 @@ public function __construct(RuleFactoryInterface $ruleFactory)
+                    $this->addRule($rule);
+                }
+
+                $this->addItem($object);
+
+                break;
+            case $object instanceof Rule:
+                /* @var $object Rule */
+                $this->addRule($object);
+                break;
+            default:
+                throw new InvalidArgumentException('Adding unsupported object type.');
         }
-
-        throw new InvalidArgumentException('Adding unsupported object type.');
     }
 
     protected function createRule($name): Rule
     {
         return $this->ruleFactory->create($name)->setName($name);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function remove($object)
+    public function remove(BaseItem $object): void
     {
-        if ($object instanceof Item) {
-            return $this->removeItem($object);
-        } elseif ($object instanceof Rule) {
-            return $this->removeRule($object);
+        switch (true) {
+            case $object instanceof Item:
+                /* @var $object Item */
+                $this->removeItem($object);
+                break;
+            case $object instanceof Rule:
+                /* @var $object Rule */
+                $this->removeRule($object);
+                break;
+            default:
+                throw new InvalidArgumentException('Removing unsupported object type.');
         }
-
-        throw new InvalidArgumentException('Removing unsupported object type.');
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function update($name, $object)
+    public function update(string $name, BaseItem $object): void
     {
-        if ($object instanceof Item) {
-            if ($object->ruleName && $this->getRule($object->ruleName) === null) {
-                $rule = $this->createRule($object->ruleName);
-                $this->addRule($rule);
-            }
-
-            return $this->updateItem($name, $object);
-        } elseif ($object instanceof Rule) {
-            return $this->updateRule($name, $object);
+        switch (true) {
+            case $object instanceof Item:
+                /* @var $object Item */
+                if ($object->ruleName && $this->getRule($object->ruleName) === null) {
+                    $rule = $this->createRule($object->ruleName);
+                    $this->addRule($rule);
+                }
+
+                $this->updateItem($name, $object);
+                break;
+            case $object instanceof Rule:
+                /* @var $object Rule */
+                $this->updateRule($name, $object);
+                break;
+            default:
+                throw new InvalidArgumentException('Updating unsupported object type.');
         }
-
-        throw new InvalidArgumentException('Updating unsupported object type.');
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getRole($name)
+    public function getRole(string $name): Role
```

Comment:
Роль может и не найтись.

```suggestion
    public function getRole(string $name): ?Role
```